app/services/markdown_processor/parser.rb

```diff
@@ -36,6 +36,7 @@ def finalize(link_attributes: {})
         html = e.message
       end
 
+      html = "<div></div>" if html.empty?
```

Comment:
If we proceed with this fix, should we change this to `if html.blank?` to catch the cases where `html` could be `nil` as well?

Reply:
That's true, errors are still coming in the `successCb` (`this.showPreview`) rather than `errorCb` (`this.failedPreview`). I guess the catch on the AJAX request only catches the error when its a connection/unexpected error rather than a response with status code other than 200

If we add something to bubble up the error [here](https://github.com/forem/forem/blob/master/app/javascript/article-form/actions.js#L16-L18) to the failureCb then the errros should render correctly in this solution @nickytonline